Added ecommerce menu and measurement integration

saurabh-pingale-cognito opened the pull request:
- Created ecommerce main menu with submenus: Categories, Sub Categories, Products, and Orders
- Built UI for all submenus and integrated backend APIs
- Implemented Create, Update, and Delete functionality for each module
- Added EcommerceMeasurement entity, service, and controller
- Established relationship between Measurement and Product entities

saurabh-pingale committed:
Added ecommerce menu and measurement integration

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-sub-categories.service.ts:
```diff
@@ -37,23 +49,30 @@ export class SubCategoriesService {
-      subCategory.country = {
-        id: updateSubCategoryDto.country_id,
-      } as Country;
+    if (country_id! in updateSubCategoryDto) {
```
don't complicate, make it simple and rewrite this logic

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/services/ecommerce-sub-categories.service.ts:
```diff
@@ -37,23 +49,30 @@ export class SubCategoriesService {
+        subCategory.country = {
+          id: updateSubCategoryDto.country_id
+        } as Country;
+      }
```
same here as well

```
 if (updateSubCategoryDto.country_id === null) {
        subCategory.country = null;
      } else if (typeof updateSubCategoryDto.country_id === 'string') {
        subCategory.country = {
          id: updateSubCategoryDto.country_id
        } as Country;
      }
```

saurabh-pingale committed:
Removed unnecessary code,
improved code, replaced discount_amount with discount_percentage in admin/backend.

saurabh-pingale committed:
Replaced discount_amount with discount_percentage and mock data with actual API in client, made category and product controllers public, added remaining fields in forms for Category, SubCategory, Product.

saurabh-pingale committed:
Removed mapper from PackageDetail and updated associated code to use data from API Response, added constant values in constants, splitted code in components

saurabh-pingale committed:
Removed mappers from the backend, added controller, service, entity and dto for shipment, removed shipment related code from the packages, added shipment UI in client and admin side.

BalanaguYashwanth commented on frontend/warehouse-app/src/types/ecommerce.ts:
```diff
@@ -38,7 +38,9 @@ export interface EcommerceProduct {
-  measurement?: string;
+  measurement?: {
+    label: string;
+  };
```
Measurement why label?

BalanaguYashwanth commented on frontend/warehouse-app/src/store/ecommerceStore.ts:
```diff
@@ -36,238 +44,30 @@ interface CartState {
+  addToCart: (productId: string, quantity?: number) => Promise<void>;
+  updateCartItem: (itemId: string, quantity: number) => Promise<void>;
+  removeFromCart: (itemId: string) => Promise<void>;
+  clearCart: () => Promise<void>;
```
We should not pass set i.e setCart 
from one prop to another prop

BalanaguYashwanth commented on frontend/warehouse-app/src/store/ecommerceStore.ts:
```diff
@@ -296,150 +96,147 @@ export const useEcommerceStore = create<EcommerceStore>()(
-            (product.description && product.description.toLowerCase().includes(searchQuery.toLowerCase()))
+          filtered = filtered.filter(
+            (product) =>
+              product.name.toLowerCase().includes(searchQuery.toLowerCase()) ||
```
This search should happen in backend, next time onwards where ever we are using search do it from backend not in frontend vis APIs

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/dto/product/create-product.dto.ts:
```diff
@@ -33,22 +35,30 @@ export class CreateEcommerceProductDto {
   @IsNumber()
   @IsPositive()
-  quantity: number;
+  unit_value: number;
```
Keep quantity instead of unity value

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/dto/sub_category/update-sub_category.dto.ts:
```diff
@@ -1,11 +1,27 @@
+  category_id: string;
+
+  @IsString()
+  @IsOptional()
```
Name can't be emptyi.e optional

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/dto/product/create-product.dto.ts:
```diff
@@ -33,22 +35,30 @@ export class CreateEcommerceProductDto {
 
   @IsString()
-  @IsNotEmpty()
+  @IsOptional()
```
Image url can't be empty

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/dto/product/create-product.dto.ts:
```diff
@@ -33,22 +35,30 @@ export class CreateEcommerceProductDto {
+  @IsBoolean()
   @IsOptional()
-  measurement: string;
+  is_active: boolean;
```
Is_activr will be compulsory but in dto and entity will define default as true

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/dto/sub_category/create-sub_category.dto.ts:
```diff
@@ -17,6 +19,7 @@ export class CreateEcommerceSubCategoryDto {
 
   @IsString()
   @IsNotEmpty()
+  @IsOptional()
```
I didn't understand this one how come not empty and optional will be together

saurabh-pingale committed:
Updated the UI, controller, service, entity and dto for the shipments.

saurabh-pingale committed:
Progressed Shipment functionality:
- Added remaining UI for Shipment Admin
- Updated code to use Shipment instead of package in Shipment Export
- Removed some shipment related code from packages
- Updated code to follow the shipment API response

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/dto/measurement/create-measurement.dto.ts:
```diff
@@ -0,0 +1,7 @@
+export class CreateEcommerceMeasurementDto {
+  @IsString()
+  @IsNotEmpty()
+  label: string;
```
Can you add label and value, I see someplace we are using value in frontend, once check if label is enough then value is not required

BalanaguYashwanth commented on backend/warehouse-server/src/ecommerce/entities/ecommerce-product.entity.ts:
```diff
@@ -42,12 +43,19 @@ export class EcommerceProduct extends BaseTimestampEntity {
-  @Column()
-  quantity: number;
+  @Column({ type: 'decimal', precision: 10, scale: 2, default: 0 })
+  unit_value: number;
```
What is the difference stock  quantity vs unit value

BalanaguYashwanth commented on frontend/warehouse-app/src/components/Header/HeaderProfileTrigger.tsx:
```diff
@@ -0,0 +1,69 @@
+      {currentUser ? (
+        <>
+          <Box>
+            <p className="px-1 py-2 text-sm font-medium rounded-md transition-all duration-200">
```
from below code of currentUser - seperate component

saurabh-pingale committed:
Fixed mentioned changes

saurabh-pingale committed:
- fixed small UI chages in shipment detail
- fixed ecommerce (add to cart, place order)
- added order in the client

saurabh-pingale committed:
Fixed Code:
-  Discount calculation
- Items getting added to cart while refresh
- Restrict add to cart till stock quantity
- If no available product then show product as "Out of Stock"
- Improve checkout page UI
- Resolve bugs from cliq history messages

saurabh-pingale committed:
Fixed bugs:
- fixed API
- fixed status color
- fixed circular structure error

saurabh-pingale committed:
Added elastic beanstalk and sh files to work with the AWS beanstalk

rohit133 committed:
Shopfront refactor -> Cart, Product Page and Home page

BalanaguYashwanth committed:
add aws elasticbeam workflow for nestjs

BalanaguYashwanth committed:
readme change to trigger github workflow

BalanaguYashwanth committed:
recorrct in github workflow yaml

BalanaguYashwanth committed:
recorrect in github workflow yaml

BalanaguYashwanth committed:
recorrect package manager pnpm in github workflow yaml

BalanaguYashwanth committed:
add pnpm.yaml file for github workflow

BalanaguYashwanth committed:
correct the port and add health api

BalanaguYashwanth committed:
add procfile for correct install in github actions

BalanaguYashwanth committed:
add nvmrc for correct deployment refernce version

BalanaguYashwanth committed:
add script with procfile for aws elastic beamstack

BalanaguYashwanth committed:
add category to the admin ecommerce section

BalanaguYashwanth committed:
revert platform files and add new github action workflows

BalanaguYashwanth committed:
moved github workflow inside the warehouse server

BalanaguYashwanth committed:
moved github workflow to root

BalanaguYashwanth committed:
add pnpm installation in workflow yml

BalanaguYashwanth committed:
correct the app name and env name in workflow.yml

BalanaguYashwanth committed:
revert package start command in nestjs warehouse server

BalanaguYashwanth committed:
add cloudinary in package backend server

BalanaguYashwanth committed:
add correct start command in server package.json file

BalanaguYashwanth committed:
add cloudinary in package json server file

BalanaguYashwanth committed:
update procfile

BalanaguYashwanth committed:
update ecommerce front main page header

BalanaguYashwanth committed:
fix workflow yml file

BalanaguYashwanth committed:
remove unuse file of zip, add only backend server files modified in workflow yml file

BalanaguYashwanth committed:
removed nooptional in action worklow yml file

BalanaguYashwanth committed:
final:update workflow yml to handle css or assets files

BalanaguYashwanth committed:
add tslib lib to package.json

BalanaguYashwanth committed:
add tslib lib to package-lock.json

BalanaguYashwanth committed:
add uid lib to server for worflow.yml to work

BalanaguYashwanth committed:
add express, load-esm and multer to server

BalanaguYashwanth committed:
add iterare to server

BalanaguYashwanth committed:
trigger to push small readme change

BalanaguYashwanth committed:
Merge pull request #50 from Cognito-Innovations/feature/shopfront-refactor

Shopfront refactor -> Cart, Product Page and Home page

BalanaguYashwanth committed:
improved skeleton loader for ecommerce page and cargo_type to seeddata and installed fast string lib for aws beamstack

saurabh-pingale committed:
Added missing libraries and added nod_modules in the .ebignore

saurabh-pingale committed:
Added loadsh dependency

saurabh-pingale committed:
Added js-yaml dependency

saurabh-pingale committed:
Added @nestjs/mapped, @nestjs/swagger, class-transfoermer and class-validator dependencies

saurabh-pingale committed:
Added ansis dependency

saurabh-pingale committed:
Added sql-highlight dependency

saurabh-pingale committed:
Added @sqltools/formatter dependency

saurabh-pingale committed:
Added app-root-path dependency

saurabh-pingale committed:
Added sha.js dependency

saurabh-pingale committed:
Added inherits dependency

saurabh-pingale committed:
Added safe-buffer dependency

saurabh-pingale committed:
Added to-buffer dependency

saurabh-pingale committed:
Added isarray dependency

saurabh-pingale committed:
Added typed-array-buffer dependency

saurabh-pingale committed:
Added es-errors/type dependency and updated .npmrc to force install all dependencies

saurabh-pingale committed:
Added call-bound dependency

saurabh-pingale committed:
Added EB platform hooks for pnpm

saurabh-pingale committed:
Added shamefully-hoist to .npmrc and updated the hook script

saurabh-pingale committed:
Fix EB hooks: correct Bash content, add cd/logging, enforce hoisting for pnpm

saurabh-pingale committed:
Added hoisted layout in .npmrc

saurabh-pingale committed:
Added get-intrinsic dependency

saurabh-pingale committed:
Removed hosting layout from .npmrc

saurabh-pingale committed:
Fixed pnpm-lock.yaml

saurabh-pingale committed:
Added get-intrinsic dependency

saurabh-pingale committed:
- Added multi select cuntries
- Fixed authentication issues on home and product details page
- Added "Login" button if user not logged; added "Add Address" button if user loggd In and no address; else show address
- Mapped categories in promtional card
- Aligned product details tab
- Fixed offer cards to show in collapse if more than one card is there else show simply card

BalanaguYashwanth committed:
update latest workfile.yml

BalanaguYashwanth committed:
Merge branch 'fix/bugs' of https://github.com/Cognito-Innovations/warehouse into fix/bugs

BalanaguYashwanth committed:
removed ignore and platform hooks file

BalanaguYashwanth committed:
correct zip path structure in workflow.yml

BalanaguYashwanth committed:
add correct zip path in workflow.yml

BalanaguYashwanth committed:
add correct pnpm install package run command in workflow.yml

BalanaguYashwanth committed:
removed prod in workflow.yml

BalanaguYashwanth committed:
revert yml file

BalanaguYashwanth committed:
update wokflow yml file

BalanaguYashwanth committed:
correct the api base url of ecommerce api

saurabh-pingale committed:
Refactord Code:
- Added cancel for removing the selected countries
- Added loder while fetching the countries in categories, sub-categories and products
- Added wrapper to return the price based on country and used in ecommerce-product service
- Added cargo-options and set up relationship with the ecommerce-categories
- Added mobile responsiveness for the ecommerce (home, products details, cart)
- Improved code

BalanaguYashwanth committed:
Merge pull request #51 from Cognito-Innovations/refactor/design-backend-admin-changes

Refactor design improvements

saurabh-pingale committed:
Refactor Code:
- Fixed scroll button for desktop screen
- Fixed image height/width in product details page
- Fixed price fluctuating in product details page
- Added "All" promotional card
- Added search functionality
- Fixed synching price between cart and checkout
- Added edit, delete and toggle for product in admin
- Replace Image URL with the Image uploader
- Improve code

BalanaguYashwanth committed:
add todods in ecommerce products

saurabh-pingale committed:
Refactor Code:
- Fixed location detection (detected by IP, fallback as geo location)
- Added infinite scrolling using Intersection Oberserver API
- Fixed API calls on Ecommerce page when redirect back
- Preserve product and categories in zustand store until page refresh
- Added "View Order History" button in Order Success Popup and redirect to order page on dashbaord
- Removed Ids in category, subcategory and products in admin
- Added rate limit using throttle
- Added IP detection middleware, entity, service and interceptor to get the IP from request, store it in DB if not exist and return in headers
- Added interceptor to attach id to each request to track IP

saurabh-pingale committed:
Added payment flow

saurabh-pingale committed:
Fixed bugs:
- Fixed skeleton loaders to wrok partially
- Fixed skeleton loader in the cart page
- Fixed add to cart, increment, decrement and delete in cart page
- Fixed cashfree payment gateway in backend and frontend

saurabh-pingale committed:
- Fixed total amount to convert by order currency before passing to cashfree
- Removed OrderSuccess Popup to display after paymant successful because we're redirecting to the order in dashbaord
- Fixed add address functionality if there is no address available

BalanaguYashwanth committed:
Merge pull request #54 from Cognito-Innovations/fix/bugs-cashfree

Fixed bugs:

BalanaguYashwanth committed:
Merge pull request #53 from Cognito-Innovations/config/cashfree

Added payment flow

BalanaguYashwanth committed:
Merge pull request #52 from Cognito-Innovations/refactor/warehouse-design-code

Refactor Code:

BalanaguYashwanth committed:
Merge pull request #49 from Cognito-Innovations/fix/bugs

Fixed bugs:
